### PR TITLE
chore: tweak tokio dependencies in `Cargo.toml`

### PR DESCRIPTION
**Pull request by carllerche**
This migrates to a better defined process for managing dependencies
between Tokio sub crates.

Refs #765

# Checklist

* [ ] Update .travis.yml to run with patches as well
* [ ] Update contributing guide.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -56,27 +56,27 @@ appveyor = { repository = "carllerche/tokio", id = "s83yxhy9qeb58va7" }
...
-tokio-fs = { version = "0.1.3", path = "tokio-fs" }
+tokio-codec = "0.1.0"
+tokio-current-thread = "0.1.3"
+tokio-io = "0.1.6"
+tokio-executor = "0.1.5"
+tokio-reactor = "0.1.1"
+tokio-threadpool = "0.1.4"
+tokio-tcp = "0.1.0"
+tokio-udp = "0.1.0"
+tokio-timer = { path = "tokio-timer" }
```
[the remark this answers is not in the document]
Comment (carllerche) [reply Cargo.toml]: @jonhoo This is versioned as such because `tokio` depends on unreleased `tokio-timer` changes.

**Comment by carllerche** [comment]
@jonhoo I agree... and the contributing doc should be updated.

**Comment by tobz** [comment]
I'm going to take a pass tonight at updating the Travis configuration to do the path-based patching.

**Comment by carllerche** [comment]
Thanks for doing this.

One issue that I haven't quite figured out how to deal with is the release process. When releasing, there can be no path based dependencies in `Cargo.toml`, but CI can't pass yet because the required versions haven't been released.

Take `tokio-timer` in this PR (though I think it has since been released?).

`tokio` uses a path dependency to depend on `tokio-timer`. When submitting the PR to do a release, this has to be changed to a version dep specifiying a version that has not yet been pushed to crates.io. This will cause CI to fail.

Do you have any thoughts on how to deal with this?

**Comment by LucioFranco** [comment]
@carllerche since `tokio-timer` already depends on a released version of `tokio-executor` then if CI detects a crate that is not published, it should publish that crate then continue.

**Comment by carllerche** [comment]
I finished this change in #991.

**Pull request closed without merging**